src/rust/src/unified_keys_ffi.rs

```diff
@@ -0,0 +1,139 @@
+use libc::c_char;
+use std::ffi::{CStr, CString};
+use tracing::error;
+
+use zcash_address::unified::{Container, Encoding, Fvk, Ufvk};
+
+use crate::address_ffi::network_from_cstr;
+
+#[no_mangle]
+pub extern "C" fn unified_full_viewing_key_free(key: *mut Ufvk) {
+    if !key.is_null() {
+        drop(unsafe { Box::from_raw(key) });
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn unified_full_viewing_key_clone(key: *const Ufvk) -> *mut Ufvk {
+    unsafe { key.as_ref() }
+        .map(|key| Box::into_raw(Box::new(key.clone())))
+        .unwrap_or(std::ptr::null_mut())
+}
+
+#[no_mangle]
+pub extern "C" fn unified_full_viewing_key_parse(
+    network: *const c_char,
+    encoded: *const c_char,
+) -> *mut Ufvk {
+    let network = match network_from_cstr(network) {
+        Some(n) => n,
+        None => {
+            return std::ptr::null_mut();
+        }
+    };
+
+    match unsafe { CStr::from_ptr(encoded) }.to_str() {
+        Ok(encoded) => match Ufvk::decode(encoded) {
+            Ok((parsed_network, fvk)) => {
+                if parsed_network == network {
+                    Box::into_raw(Box::new(fvk))
+                } else {
+                    error!(
+                        "Key was encoded for a different network ({:?}) than what was requested ({:?})",
+                        parsed_network,
+                        network,
+                    );
+                    std::ptr::null_mut()
+                }
+            }
```

Comment:
Minor style nit: this could be refactored into (modulo `cargo fmt`):
```suggestion
            Ok((parsed_network, fvk)) if parsed_network == network => {
                Box::into_raw(Box::new(fvk))
            }
            Ok((parsed_network, _)) => {
                error!(
                    "Key was encoded for a different network ({:?}) than what was requested ({:?})",
                    parsed_network,
                    network,
                );
                std::ptr::null_mut()
            }
```
(I have a slight preference for reducing nesting levels.)